src/orquestra/sdk/_base/abc.py

```diff
@@ -15,60 +15,27 @@
 from datetime import timedelta
 from pathlib import Path
 
-from orquestra.sdk._base._spaces._structs import Project, ProjectRef, Workspace
-from orquestra.sdk.exceptions import WorkspacesNotSupportedError
-from orquestra.sdk.schema.configs import RuntimeConfiguration
-from orquestra.sdk.schema.ir import TaskInvocationId, WorkflowDef
-from orquestra.sdk.schema.local_database import StoredWorkflowRun
-from orquestra.sdk.schema.responses import WorkflowResult
-from orquestra.sdk.schema.workflow_run import (
+from ..exceptions import WorkspacesNotSupportedError
+from ..schema.configs import RuntimeConfiguration
+from ..schema.ir import TaskInvocationId, WorkflowDef
+from ..schema.local_database import StoredWorkflowRun
+from ..schema.responses import WorkflowResult
+from ..schema.workflow_run import (
     ProjectId,
     State,
     WorkflowRun,
     WorkflowRunId,
     WorkflowRunMinimal,
     WorkspaceId,
 )
-
-
-class LogReader(t.Protocol):
-    """
-    A component that reads logs produced by tasks and workflows.
-    """
-
-    def get_task_logs(
-        self, wf_run_id: WorkflowRunId, task_inv_id: TaskInvocationId
-    ) -> t.List[str]:
-        """
-        Reads all available logs, specific to a single task invocation/run.
-
-        Returns:
-            Log lines printed when running this task invocation. If the task didn't
-            produce any logs this should be an empty list.
-        """
-        ...
-
-    def get_workflow_logs(
-        self, wf_run_id: WorkflowRunId
-    ) -> t.Dict[TaskInvocationId, t.List[str]]:
-        """
-        Reads all available logs printed during execution of this workflow run.
-
-        Returns:
-            A mapping with task logs. Each key-value pair corresponds to one task
-            invocation.
-            - key: task invocation ID (see
-                orquestra.sdk._base.ir.WorkflowDef.task_invocations)
-            - value: log lines from running this task invocation
-        """
-        ...
-
+from ._logs._interfaces import LogReader, WorkflowLogs
+from ._spaces._structs import Project, ProjectRef, Workspace
 
 # A typealias that hints where we expect raw artifact values.
 ArtifactValue = t.Any
 
 
-class RuntimeInterface(ABC):
+class RuntimeInterface(ABC, LogReader):
```

Comment:
inb4: I wanted to keep `RuntimeInterface` and `LogReader` separate but there are places where we return a single object that conforms to both interfaces, like a `foo: RuntimeInterface & LogReader`. The way to specify that in Python is to define a single class that inherits from both supertypes ([see the discussion on SO](https://stackoverflow.com/questions/60732924/how-to-specify-that-a-python-object-must-be-two-types-at-once)).

Reply:
Instead of having a third type like `class FullRuntime(RuntimeInterface, LogReader): pass` I went with single inheritance here. I think it'll be the way to go if we change  `RuntimeInterface` into smaller interfaces specified as protocols instead of ABCs (I think that's our long term plan, right?).

Reply:
I think this is OK, but we should be careful.

In this case, this just makes sure `RuntimeInterface` conforms to the `LogReader` protocol through type checking.

> The way to specify that in Python is to define a single class that inherits from both supertypes ([see the discussion on SO](https://stackoverflow.com/questions/60732924/how-to-specify-that-a-python-object-must-be-two-types-at-once)).

This is slightly different, this is about making a new type only applies for mixing two protocols to make a 3rd super-protocol. What we have above is saying `RuntimeInterface` needs to implement `LogReader` but in the SO question they're mixing two protocols.